src/App.js

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog'
+
 
 const App = () => {
+  const [chatData, setChatmessages] = useState(chatMessages)
+
+  const updateChatData = updatedChat => {
+    const chats = chatData.map(chat => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
+      } else {
+        return chat;
+      }
+    });
+
+    setChatmessages(chats);
+  };
+
+  const calcTotalLikes = (chatData) => {
+    return chatData.reduce((total, chat) => {
+      return total + chat.liked;
```

Comment:
Way to use reduce here!

It's cool that you leveraged the knowledge that in JavaScript if you add a boolean to something it gets cast as a number (0 for false, 1 for true). Prefer you use conditional logic to add a number to a number instead.